src/Workspaces/Remote/ServiceHub/Shared/RoslynJsonConverter.cs

```diff
@@ -55,11 +56,12 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
         private static void Add<T>(
             ImmutableDictionary<Type, JsonConverter>.Builder builder,
             BaseJsonConverter<T> converter)
-        {
-            builder.Add(typeof(T), converter);
-        }
+            => builder.Add(typeof(T), converter);
```

Comment:
@heejaechang I ended up leaving the immutable dictionary creation for well-known converters because of the above `AppendRoslynSpecificJsonConverters` partial method which have other external consumers. Instead I just exposed the base type and a new `TryAdd` method...

Reply:
@shyamnamboodiripad that's fine. was there a specific reason why you split it into a base and derived type? otherwise, I think we can leave it as 1 type and do the conversion you did.

Reply:
I have reverted the internal modifier change fore the base type. I had added that so that the Add method can be more strong typed. But I have changed the add method to accept a regular JsonConverter - exposing the base type didn't  help since there are multiple copies of this type (in language services and remote service hub projects).